testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/BaseTestDeltaLakeHdfsReads.java

```diff
@@ -76,6 +77,6 @@ public void testReads()
                 "WITH (location = 'hdfs://hadoop-master:9000/tmp/region')");
 
         assertThat(onTrino().executeQuery("SELECT count(*) FROM delta.default.region")).containsOnly(row(5L));
-        onTrino().executeQuery("DROP TABLE delta.default.region");
+        retryOnModifiedConcurrentlyFailure(() -> onTrino().executeQuery("DROP TABLE delta.default.region"));
```

Comment:
As discussed offline.

if this is a situation a user can legitimately run into, retrying in tests will only cover up actual product problem, preventing us from fixing it.

Reply:
That's why I didn't add the retry in `TestDeltaLakeDropTableCompatibility`. I can revert all changes for `onTrino` though.

Reply:
> I can revert all changes for `onTrino` though.

👍

Reply:
Reverted.

Reply:
Unfortunately, CI failed when dropping a table on Trino. 
https://github.com/trinodb/trino/actions/runs/3569815668/jobs/6000754659
```
tests               | 	Suppressed: java.lang.Exception: Query: DROP TABLE delta.default.test_compression_GZIP_7lrqddrubi
```